src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.CSharp.targets

```diff
@@ -38,4 +38,13 @@ Copyright (c) .NET Foundation. All rights reserved.
   <ItemGroup Condition="'$(SupportsHotReload)' != 'false' AND '$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '6.0'))">
     <ProjectCapability Include="SupportsHotReload" />
   </ItemGroup>
+
+  <!--
+    BinaryFormatter infrastructure is obsolete as error in 7.0+.
+    When https://github.com/Microsoft/visualfsharp/issues/3207 is fixed,
+    remove the block below and move it into the shared .targets file.
+  -->
+  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '7.0'))">
+    <WarningsAsErrors Condition="'$(EnableUnsafeBinaryFormatterSerialization)' != 'true'">$(WarningsAsErrors);SYSLIB0011</WarningsAsErrors>
```

Comment:
Why is this testing for `!= true`? Based on the property name, if the formatter isn't enabled, the value would be false (not enabled).

Reply:
The default value is "" (not set) or "false", depending on target SDK. We want both of them to result in compile-time errors. We can't default the value to "false" in all cases just yet because doing so also affects runtime behavior, which we don't want to do. The full matrix is listed at https://github.com/dotnet/runtime/issues/72132.